compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1104,14 +1104,14 @@ fn link_natively(
         let stripcmd = "rust-objcopy";
         match (strip, crate_type) {
             (Strip::Debuginfo, _) => {
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &["-S"])
+                strip_with_external_utility(sess, stripcmd, out_filename, &["--strip-debug"])
             }
             // Per the manpage, `-x` is the maximum safe strip level for dynamic libraries. (#93988)
             (Strip::Symbols, CrateType::Dylib | CrateType::Cdylib | CrateType::ProcMacro) => {
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &["-x"])
+                strip_with_external_utility(sess, stripcmd, out_filename, &["-x"])
```

Comment:
Suggestion [DISCARD-ALL 1/2]: can we spell out the full flag for this as well? `llvm-objcopy` docs say that `-x` is

> --discard-all, -x

>   Remove most local symbols from the output. Different file formats may limit this to a subset of the local symbols. For example, file and section symbols in ELF objects will not be discarded. Additionally, remove all debug sections.

This happens to align with `llvm-strip`, unlike `-S` or no-flag.

- https://llvm.org/docs/CommandGuide/llvm-strip.html#cmdoption-llvm-strip-discard-all
- https://llvm.org/docs/CommandGuide/llvm-objcopy.html#cmdoption-llvm-objcopy-discard-all